tests/components/bsblan/test_climate.py

```diff
@@ -0,0 +1,186 @@
+"""Tests for the BSBLAN climate component."""
+
+from unittest.mock import AsyncMock, MagicMock
+
+from bsblan import BSBLANError, State
+import pytest
+
+from homeassistant.components.bsblan.climate import BSBLANClimate
+from homeassistant.components.bsblan.coordinator import BSBLanCoordinatorData
+from homeassistant.components.climate import PRESET_ECO, PRESET_NONE, HVACMode
+from homeassistant.const import UnitOfTemperature
+from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
+
+
+class MockStateValue:
+    """Mock class for a BSBLan State value."""
+
+    def __init__(self, value):
+        """Initialize the mock class."""
+        self.value = value
+
+
+@pytest.fixture
+def mock_state():
+    """Create a mock BSBLan State object."""
+    state = MagicMock(spec=State)
+    state.current_temperature = MockStateValue("22.5")
+    state.target_temperature = MockStateValue("21.0")
+    state.hvac_mode = MockStateValue("heat")
+    return state
```

Comment:
Rather just move stuff to conftest and make a mock you use for the whole integration tests instead of climate specifically